### DISCUSSION TITLE
A kernel for calculating the residual and Jacobian contributions from a function

### DISCUSSION THREAD

**Discussion by alimostafavi24** [idaholab/moose#28124] [Q&A Modules: General]
Dear MOOSE experts,

Is there any kernel to calculate the residual and Jacobian contributions from a given function (f) in the framework?

I am looking for something similar to SplitCHParsed where it calculates the contributions from df/dc. I need something that does not take the derivative of the function.

Thanks,

**Comment by GiudGiud** [comment]
Hello

Do you mean a CoupledForce like kernel which takes a Function as an input?

Or do you mean an example of any kernel using a Function in the code?

Guillaume

**Reply by alimostafavi24** [reply]
Hello,

Sorry for the late reply. I am mostly looking for a CoupledForce like kernel which takes a Function as an input and computes the resiudal and jacobian contributions for a given function.

Thanks,
Ali

**Reply by GiudGiud** [reply]
forward the function through an auxvariable  with a FunctionAux
then use this kernel
https://mooseframework.inl.gov/source/kernels/CoupledForce.html
